crates/astria-sequencer/src/transaction/signed_transaction.rs

```diff
@@ -0,0 +1,155 @@
+use anyhow::{
+    anyhow,
+    bail,
+    Context as _,
+    Result,
+};
+use astria_proto::sequencer::v1::{
+    unsigned_transaction::Value::AccountsTransaction as ProtoAccountsTransaction,
+    SignedTransaction as ProtoSignedTransaction,
+    UnsignedTransaction as ProtoUnsignedTransaction,
+};
+use async_trait::async_trait;
+use penumbra_storage::{
+    StateRead,
+    StateWrite,
+};
+use prost::Message as _;
+use tracing::instrument;
+
+use crate::{
+    accounts::{
+        transaction::Transaction as AccountsTransaction,
+        types::Address,
+    },
+    crypto::{
+        PublicKey,
+        Signature,
+        Verifier,
+    },
+    hash,
+    transaction::{
+        ActionHandler,
+        TransactionHash,
+        UnsignedTransaction,
+    },
+};
+
+#[derive(Debug, Clone)]
+pub struct SignedTransaction {
+    pub signature: Signature,
+    pub public_key: PublicKey,
+    pub transaction: UnsignedTransaction,
+}
+
+impl SignedTransaction {
+    /// Verifies the transaction signature.
+    /// The transaction signature message is the hash of the transaction.
+    ///
+    /// # Errors
+    ///
+    /// - If the signature is invalid
+    pub fn verify_signature(&self) -> Result<()> {
+        self.public_key
+            .verify(&self.transaction.hash(), &self.signature)
+            .context("failed to verify transaction signature")
+    }
+
+    /// Returns the address which signed the transaction.
+    ///
+    /// # Errors
+    ///
+    /// - If the public key cannot be converted into an address
+    pub fn signer_address(&self) -> Result<Address> {
+        Address::try_from(&self.public_key)
+    }
+
+    #[must_use]
+    pub fn to_vec(&self) -> Vec<u8> {
```

Comment:
The fact that `to_vec` here (and for the types) actually serializes to protobuf is very surprising and unexpected. 

From a first glance it also looks like these `to_vec` are only used in `hash` calls`?

Instead of these indirections, please just make it explicit at the call site: prefer something like `Sha256::hash(signed_transaction.to_proto())

Reply:
this brings up a question; how should we serialize transactions that are sent over the network/RPC? protobuf or borsh? i assumed protobuf for now since that's sort of cosmos-standard but we don't actually need to stick with that.

Reply:
I'd prefer we stick to what is more "standard" as well as just minimizing the number of serialization formats we use across our stack. Looking at [performance comparisons](https://github.com/djkoloski/rust_serialization_benchmark), it looks like borsch would save us some time at the cost of space, saving over wire bandwidth, is probably worthwhile is my gut feel.